src/timeMachine/actions/queries.ts

```diff
@@ -299,7 +299,10 @@ export const executeQueries = (abortController?: AbortController) => async (
       if (isCurrentPageDashboard(state)) {
         // reset any existing matching query in the cache
         resetQueryCacheByQuery(text)
-        const result = getCachedResultsOrRunQuery(orgID, text, state)
+        const usedVars = filterUnusedVarsBasedOnQuery(getAllVariables(state), [
```

Comment:
seems like this needs to go inside of `getCachedResultsOrRunQuery` if you're making copy pasta with it. the interface being `nameThingsBetter(orgID, text, getAllVariables(state))` and having the function filter out any of those variables not referenced in the query before making the hash

Reply:
I see what you're saying. So we should be passing all the variables and then filtering them within the child component? What's the benefit of that as opposed to just determining what's used in the parent, and passing it along